src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +30,246 @@ const ActorInit Dm_Zl_InitVars = {
+
+            if (nextAnimationIndex != this->animationIndex) {
+                this->animationIndex = nextAnimationIndex;
+                DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], ZELDA_ANIM_FACING_AWAY);
+            }
+        }
+
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) != 0) {
+        actionIndex = this->animationIndex;
+        if ((actionIndex == 1) || (actionIndex == 3) || (actionIndex == 5)) {
+            this->animationIndex++;
+            DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], ZELDA_ANIM_FACING_AWAY);
+        }
+    }
+}
+
+/**
+ * Updates the eye blinking and state, and mouth textures.
+ *  (Assumption: these modifications are from outside of this actor by a cutscene).
+ */
+void DmZl_UpdateFace(DmZl* this) {
+    if (this->blinkTimer > 0) {
+        this->blinkTimer--;
+    } else {
+        this->blinkTimer = 0;
+    }
+
+    if (this->blinkTimer < 3) {
+        this->eyeTextureIndexRight = this->blinkTimer;
+        this->eyeTextureIndexLeft = this->blinkTimer;
+    }
+
+    switch (this->nextEyeState) {
+        case 0:
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+        case 1:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_CLOSED;
+            }
+            break;
+        case 2:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = ZELDA_EYE_OPEN_LOOKING_LEFT;
+                this->eyeTextureIndexRight = ZELDA_EYE_OPEN_LOOKING_RIGHT;
+            }
+            break;
+        case 3:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = ZELDA_EYE_OPEN_LOOKING_RIGHT;
+                this->eyeTextureIndexRight = ZELDA_EYE_OPEN_LOOKING_LEFT;
+            }
+            break;
+        case 4:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_WIDE;
+            }
+            break;
+        case 5:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_HAPPY;
+            }
+            break;
+        case 6:
+            if (this->blinkTimer >= 3) {
+                this->blinkTimer = 0;
+            }
+            break;
+    }
+
+    switch (this->nextMouthState) {
+        default:
+            this->mouthTextureIndex = ZELDA_MOUTH_NEUTRAL;
+            break;
+        case 1:
+            this->mouthTextureIndex = ZELDA_MOUTH_SMILING;
+            break;
+        case 2:
+            this->mouthTextureIndex = ZELDA_MOUTH_FROWNING;
+            break;
+        case 3:
+            this->mouthTextureIndex = ZELDA_MOUTH_OPEN;
+            break;
+    }
```

Comment:
Both switches in this function can use the appropriate enums for their cases.

Reply:
I'm not convinced, if it was a one to one conversion there would be no need for a switch, they could just assign `this->mouthTextureIndex = this->nextMouthState;`

But sure I can change it.